content/terraform-docs-common/docs/cloud-docs/registry/publish-modules.mdx

```diff
@@ -5,123 +5,198 @@ description: Use the HCP Terraform private registry to publish and share private
+   - In the **Source Directory** field, specify the path to the module source code.
+
+  When **Branch** is enabled:
+
+   - In the **Branch Name** field, specify the name of the branch to scan for the module code.
+   - In the **Module Version** field, specify the version of the module.
+   - In the **Source Directory** field, specify the path to the module source code.
+
+1. Click **Next** to advance to the **Confirm selection** screen.
+1. In the **Module Name** field, specify a name for the module. This is the name that appears in the registry. HCP Terraform and Terraform Enterprise use the name in the module address. 
+1. In the **Provider Name** field, specify the name of the provider that integrates the module. HCP Terraform and Terraform Enterprise use the name in the module address.  
+1. It is optional, but if this module is a [no-code ready module](/terraform/cloud-docs/no-code-provisioning/module-design), enable the **Add Module to no-code provision allowlist** option to allow your organization members to provision resources defined by this module using the no-code workflow. Refer to [Provision no-code infrastructure](/terraform/cloud-docs/no-code-provisioning/provisioning) for more information.
+
+<!-- BEGIN: TFC:only name:pnp-callout -->
+@include 'tfc-package-callouts/nocode.mdx'
+<!-- END: TFC:only name:pnp-callout -->
+
+1. Click **Publish module**.
+
+HCP Terraform and Terraform Enterprise displays a loading page while it imports the module versions and then directs you to the new module's details page. On the details page, you can view available versions, read documentation, and copy a usage example.
+
 ### Tag-based publishing considerations
 
-When using the **Tag** module publishing type, the registry uses `x.y.z` formatted release tags to identify module versions. Your repository must contain at least one release tag for you to publish a module. Release tag names must be a [semantic version](http://semver.org), which you can optionally prefix with a `v`. For example, `v1.0.4` and `0.9.2`. The registry ignores tags that do not match these formats.
+To enable the **Tag** module publishing type, your registry must use the `x.y.z` format for release tags to identify module versions. Your repository must contain at least one release tag for you to publish a module. Release tag names must be a [semantic version](http://semver.org), which you can optionally prefix with a `v`. For example, `v1.0.4` and `0.9.2`. The registry ignores tags that do not match these formats.
 
-<!-- BEGIN: TFC:only name:branch-based-publishing -->
+To publish modules from a single repository, you should also include a tag prefix so that HCP Terraform can publish individual modules in the repository when you create new version tags. Refer to [Release tag prefixes](#release-tag-prefixes) for more information.
```

Comment:
```suggestion
To publish modules from a single repository, you should also include a tag prefix so that HCP Terraform can publish individual modules in the repository when you create new version tags. Refer to [Release tag prefixes](#release-tag-prefixes) for more information.
```

"should" or "must"?